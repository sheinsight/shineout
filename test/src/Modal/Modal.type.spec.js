import { Modal } from 'shineout'

describe('Modal[Type]', () => {
  test('should render different type', () => {
    ;['info', 'success', 'warn', 'error'].forEach(type => {
      document.body.innerHTML = ''
      Modal[type]({
        title: `This is a ${type} message`,
        content: 'this is some information',
      })
      expect(document.body.innerHTML.replace(/id=".*"/g, 'id="a')).toMatchSnapshot()
    })
  })
})
