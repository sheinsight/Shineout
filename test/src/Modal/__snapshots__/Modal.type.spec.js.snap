// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal[Type] should render different type 1`] = `"<div class=\\"so-modal\\" style=\\"display: block; background: rgba(0, 0, 0, 0.25);\\"><div class=\\"so-modal-mask\\"></div><div class=\\"so-card so-card-shadow so-modal-panel so-modal-info\\" style=\\"position: relative; display: inline-flex; width: 420px; top: 10vh;\\"><a class=\\"so-modal-close\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M602.512147 511.99738l402.747939-402.747939a63.999673 63.999673 0 0 0-90.509537-90.509537L512.00261 421.487843 109.254671 18.749904a63.999673 63.999673 0 0 0-90.509537 90.509537L421.493073 511.99738 18.755134 914.745319a63.999673 63.999673 0 0 0 90.509537 90.509537L512.00261 602.506917l402.747939 402.747939a63.999673 63.999673 0 0 0 90.509537-90.509537z\\"></path></svg></a><div class=\\"so-card-body so-modal-body\\"><div class=\\"so-modal-icon\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M512 1024c282.771 0 512-229.23 512-512s-229.23-512-512-512-512 229.23-512 512 229.23 512 512 512zM432 256c0-44.183 35.817-80 80-80s80 35.817 80 80v31.999c0 44.183-35.817 80-80 80s-80-35.817-80-80v-31.999zM431.999 512c0-44.183 35.817-80 80-80s80 35.817 80 80v256c0 44.183-35.817 80-80 80s-80-35.817-80-80v-256z\\"></path></svg></div><div class=\\"so-modal-title\\">This is a info message</div><div>this is some information</div></div><div class=\\"so-card-footer so-card-right so-modal-footer\\"><button id=\\"a><span>Ok</span></button></div></div></div>"`;

exports[`Modal[Type] should render different type 2`] = `"<div class=\\"so-modal\\" style=\\"display: block; background: rgba(0, 0, 0, 0.01);\\"><div class=\\"so-modal-mask\\"></div><div class=\\"so-card so-card-shadow so-modal-panel so-modal-success\\" style=\\"position: relative; display: inline-flex; width: 420px; top: 10vh;\\"><a class=\\"so-modal-close\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M602.512147 511.99738l402.747939-402.747939a63.999673 63.999673 0 0 0-90.509537-90.509537L512.00261 421.487843 109.254671 18.749904a63.999673 63.999673 0 0 0-90.509537 90.509537L421.493073 511.99738 18.755134 914.745319a63.999673 63.999673 0 0 0 90.509537 90.509537L512.00261 602.506917l402.747939 402.747939a63.999673 63.999673 0 0 0 90.509537-90.509537z\\"></path></svg></a><div class=\\"so-card-body so-modal-body\\"><div class=\\"so-modal-icon\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M874 150C674.09-50 349.91-50 150 150s-200 524.09 0 724 524.09 200 724 0 200-524.09 0-724zM760.57 440.57l-256 256a80 80 0 0 1-113.14 0l-128-128a80 80 0 0 1 113.14-113.14L448 526.86l199.43-199.43a80 80 0 0 1 113.14 113.14z\\"></path></svg></div><div class=\\"so-modal-title\\">This is a success message</div><div>this is some information</div></div><div class=\\"so-card-footer so-card-right so-modal-footer\\"><button id=\\"a><span>Ok</span></button></div></div></div>"`;

exports[`Modal[Type] should render different type 3`] = `"<div class=\\"so-modal\\" style=\\"display: block; background: rgba(0, 0, 0, 0.01);\\"><div class=\\"so-modal-mask\\"></div><div class=\\"so-card so-card-shadow so-modal-panel so-modal-warning\\" style=\\"position: relative; display: inline-flex; width: 420px; top: 10vh;\\"><a class=\\"so-modal-close\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M602.512147 511.99738l402.747939-402.747939a63.999673 63.999673 0 0 0-90.509537-90.509537L512.00261 421.487843 109.254671 18.749904a63.999673 63.999673 0 0 0-90.509537 90.509537L421.493073 511.99738 18.755134 914.745319a63.999673 63.999673 0 0 0 90.509537 90.509537L512.00261 602.506917l402.747939 402.747939a63.999673 63.999673 0 0 0 90.509537-90.509537z\\"></path></svg></a><div class=\\"so-card-body so-modal-body\\"><div class=\\"so-modal-icon\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M512 0C229.23 0 0 229.23 0 512s229.23 512 512 512 512-229.23 512-512S794.77 0 512 0z m80 768a80 80 0 0 1-160 0v-32a80 80 0 0 1 160 0v32z m0-256a80 80 0 0 1-160 0V256a80 80 0 0 1 160 0v256z\\"></path></svg></div><div class=\\"so-modal-title\\">This is a warn message</div><div>this is some information</div></div><div class=\\"so-card-footer so-card-right so-modal-footer\\"><button id=\\"a><span>Ok</span></button></div></div></div>"`;

exports[`Modal[Type] should render different type 4`] = `"<div class=\\"so-modal\\" style=\\"display: block; background: rgba(0, 0, 0, 0.01);\\"><div class=\\"so-modal-mask\\"></div><div class=\\"so-card so-card-shadow so-modal-panel so-modal-error\\" style=\\"position: relative; display: inline-flex; width: 420px; top: 10vh;\\"><a class=\\"so-modal-close\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M602.512147 511.99738l402.747939-402.747939a63.999673 63.999673 0 0 0-90.509537-90.509537L512.00261 421.487843 109.254671 18.749904a63.999673 63.999673 0 0 0-90.509537 90.509537L421.493073 511.99738 18.755134 914.745319a63.999673 63.999673 0 0 0 90.509537 90.509537L512.00261 602.506917l402.747939 402.747939a63.999673 63.999673 0 0 0 90.509537-90.509537z\\"></path></svg></a><div class=\\"so-card-body so-modal-body\\"><div class=\\"so-modal-icon\\"><svg viewBox=\\"0 0 1024 1024\\"><path d=\\"M512 0C232 0 5 226.64 4.97250989 506.66666667s226.99620754 507.02749011 507.02749011 507.02749011 507.02749011-226.99620754 507.02749011-507.02749011S792.03128257-0.36082344 512-0.36082344zM701.01984777 685.92623566c-22.86694019 22.84158802-59.9052983 22.84158802-82.74688632 0l-96.51268267-96.51268267-96.51268266 96.51268267c-22.86694019 22.84158802-59.9052983 22.84158802-82.74688632 0-22.84158802-22.86694019-22.84158802-59.9052983 0-82.74688632L439.01339246 506.66666667l-96.53803365-96.51268267c-22.84158802-22.86694019-22.84158802-59.9052983 0-82.74688632 22.86694019-22.84158802 59.9052983-22.84158802 82.74688632 0l96.51268267 96.51268267 96.51268266-96.51268267c22.86694019-22.84158802 59.9052983-22.86694019 82.74688633 0 22.84158802 22.86694019 22.84158802 59.9052983 0 82.74688632L604.48181413 506.66666667l96.53803364 96.51268267C723.86143696 626.04628952 723.88678795 663.08464763 701.01984777 685.92623566z\\"></path></svg></div><div class=\\"so-modal-title\\">This is a error message</div><div>this is some information</div></div><div class=\\"so-card-footer so-card-right so-modal-footer\\"><button id=\\"a><span>Ok</span></button></div></div></div>"`;
